packages/fxa-payments-server/src/routes/Subscriptions/PaymentUpdateFormV2.tsx

```diff
@@ -0,0 +1,308 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useCallback, useState } from 'react';
+import { Localized } from '@fluent/react';
+import dayjs from 'dayjs';
+import {
+  getLocalizedDateString,
+  getLocalizedDate,
+  getLocalizedCurrency,
+  formatPlanPricing,
+} from '../../lib/formats';
+import { useNonce } from '../../lib/hooks';
+import { useBooleanState } from 'fxa-react/lib/hooks';
+import { getErrorMessage } from '../../lib/errors';
+import { Stripe, StripeCardElement, StripeError } from '@stripe/stripe-js';
+import {
+  Customer,
+  CustomerSubscription,
+  Plan,
+  PlanInterval,
+} from '../../store/types';
+import { metadataFromPlan } from 'fxa-shared/subscriptions/metadata';
+import AlertBar from '../../components/AlertBar';
+import { ReactComponent as CloseIcon } from 'fxa-react/images/close.svg';
+import PaymentForm, { PaymentFormProps } from '../../components/PaymentFormV2';
+import ErrorMessage from '../../components/ErrorMessage';
+import * as Amplitude from '../../lib/amplitude';
+import * as apiClient from 'fxa-payments-server/src/lib/apiClient';
+
+type PaymentUpdateError = undefined | StripeError;
+
+export type PaymentUpdateStripeAPIs = Pick<Stripe, 'confirmCardSetup'>;
+
+export type PaymentUpdateAuthServerAPIs = Pick<
+  typeof apiClient,
+  'apiCreateSetupIntent' | 'apiUpdateDefaultPaymentMethod'
+>;
+
+export type PaymentUpdateFormProps = {
+  customer: Customer;
+  customerSubscription: CustomerSubscription;
+  plan: Plan;
+  refreshSubscriptions: () => void;
+  setUpdatePaymentIsSuccess: () => void;
```

Comment:
Needing to plumb in these `{set,reset}UpdatePaymentIsSuccess` functions is annoying. But, it seems to be the only way to support the "Your billing information has been updated successfully" banner on the subscription page. Rendering the AlertBar locally here ends up never showing it, because this component gets totally replaced on a customer refresh.